File: Introduction-To-Algorithm-Fourth-Edition/02-Getting-Started/2-1 Insertion-sort/Code-of-insertion-sort/insertion-sort.js
function insertionSort(arr, n) {
  for (let i = 1; i < n; i++) {
    let j = i - 1;
    let key = arr[i];
    while (j >= 0 && arr[j] > key) {
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = arr[j];
  }
}

File: Introduction-To-Algorithm-Fourth-Edition/02-Getting-Started/2-1 Insertion-sort/Code-of-insertion-sort/insertion-sort.cpp
#include<bits/stdc++.h>
using namespace std;
#define int long long
#define endl "\n"
void insertionSort(vector<int> &v)
{
    for(int i=1;i<v.size();i++)
    {
        int j=i-1;
        int key=v[i];
        while(j>=0 && v[j]>key)
        {
            v[j+1]=v[j];
            j--;
        }
        v[j+1]=key;
    }
}
signed main()
{
    int n;
    cin>>n;
    vector<int> v(n);
    for(auto &x:v)
        cin>>x;
    insertionSort(v);
    for(auto x:v)
        cout<<x<<" ";
    cout<<endl;
}